src/utils/search_pypi.ts

```diff
@@ -0,0 +1,70 @@
+interface PyPIPackageResult {
+  name: string;
+  package_manager_url: string;
+}
+
+let cachedData: { projects: { name: string }[] } = { projects: [] };
+
+async function fetchPyPIProjects(): Promise<{ projects: { name: string }[] }> {
+  if (cachedData.projects.length > 0) {
+    return cachedData;
+  }
+
+  const response = await fetch("https://pypi.org/simple/", {
+    headers: { Accept: "application/vnd.pypi.simple.v1+json" },
+  });
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  cachedData = await response.json();
+  return cachedData;
+}
+
+function calculateRelevanceScore(query: string, packageName: string): number {
+  const normalizedQuery = query.toLowerCase();
+  const normalizedPackageName = packageName.toLowerCase();
+
+  if (normalizedQuery === normalizedPackageName) {
+    return 1;
+  } else if (normalizedPackageName.startsWith(normalizedQuery)) {
+    return 0.9;
+  } else if (normalizedPackageName.includes(normalizedQuery)) {
+    const position = normalizedPackageName.indexOf(normalizedQuery);
+    return 0.7 - position * 0.01;
```

Comment:
Is this a standard string search algo? if not, can you either :

1) store the package names in duckdb or sqllight? 
2) maybe look at https://www.fusejs.io/ or https://www.npmjs.com/package/string-similarity-js or similar